test/test_helper.rb

```diff
@@ -7,3 +7,20 @@
 require "minispec-metadata"
 
 require "phashion" unless RUBY_ENGINE == "jruby"
+
+class MiniTest::Test
+
+  def fixture_image(name)
+    File.open("test/fixtures/#{name}")
+  end
+
+  # Creates a copy of the file and stores it into a Tempfile. Works for any
+  # IO object that responds to `#read(length = nil, outbuf = nil)`.
+  def copy_to_tempfile(file)
+    extension = File.extname(file.path) if file.respond_to?(:path)
+    tempfile = Tempfile.new(["test", extension.to_s], binmode: true)
+    IO.copy_stream(file, tempfile.path)
+    file.rewind
+    tempfile
```

Comment:
I'm guessing you get an "empty file" from vips because the file descriptor got stale, meaning it didn't pick up the changes to the underlying file from `IO.copy_stream`, you'll probably need to reopen the file with `tempfile.open` after `IO.copy_stream`. Let's write it like this:

```rb
    IO.copy_stream(file, tempfile)
    file.rewind
    tempfile.tap(&:open)
```